### PR TITLE
[prototype only] ports vite

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
<details>
<summary>what the fuck is vite</summary>

vite is a newer build tool that is written in Go. it yields significant performance gains over webpack. for instance, over the traditional react application (create-react-app), it shaves dev build speed from ~13 seconds down to sub-second times. it's so fast that it started a war to create the fastest build tools for web dev. even the creator of webpack chimed in to write turbopack in rust. vite basically deprecated create-react-app as the web standard for scaffolding react applications within a week.

outside of performance, its all-inclusive architecture means that we no longer have to include babel, webpack, and all of its plugins. the config file is stupidly easy and its polyfills for ie11 are managed within @vitejs/plugin-legacy. 

</details>

## Why It's Good For The Game
Catapults tgui build tools into 2023. Significantly faster
## Changelog
N/A

### CODE REVIEW

**Comment by Mothblocks** [comment]
Let's split js -> jsx into its own PR.

1. If this PR doesn't work out, we will want that regardless
2. It makes reading this PR significantly easier

**Comment by MrStonedOne** [comment]
Would you mind explaining wtf vite is in the pr body?

**Comment by MrStonedOne** [comment]
Also you can not use age, "legacy" or "modern" to justify a change. you have to actually spell out why it its better besides "its newer"

File: dependencies.sh [hunk 1]
```diff
@@ -11,8 +11,8 @@ export BYOND_MINOR=1588
 export RUST_G_VERSION=3.0.0
 
 #node version
-export NODE_VERSION=14
-export NODE_VERSION_PRECISE=14.16.1
+export NODE_VERSION=20
+export NODE_VERSION_PRECISE=20.9.0
```
Comment (MrStonedOne) [inline dependencies.sh:15]: Node has to be be version 14 for windows 7 reasons.

**Review by MrsTonedOne — CHANGES_REQUESTED** [review]
vite dropped support for node 14 and 15 in version 3.0 and node 14 is the highest we can go to and still support windows 7. (maybe node 15 but it requires some hacks.)

Until the client and all dev tools work on linux this PR is dead on arrival unless you want to make it work with vite 2

**Comment by jlsnow301** [comment]
> Would you mind explaining wtf vite is in the pr body?
> Also you can not use age, "legacy" or "modern" to justify a change. you have to actually spell out why it its better besides "its newer"

sure thing mso


<details>
<summary>what the fuck is vite</summary>

vite is a newer build tool that is written in Go. it yields significant performance gains over webpack. for instance, over the traditional react application (create-react-app), it shaves dev build speed from ~13 seconds down to sub-second times. it's so fast that it started a war to create the fastest build tools for web dev. even the creator of webpack chimed in to write turbopack in rust. vite basically deprecated create-react-app as the web standard for scaffolding react applications within a week.

outside of performance, its all-inclusive architecture means that we no longer have to include babel, webpack, and all of its plugins. the config file is stupidly easy and its polyfills for ie11 are managed within @vitejs/plugin-legacy. 

</details>

**Comment by jlsnow301** [comment]
i might have something that works with both

**Pull request closed without merging**